### ISSUE TITLE
Steam game doesn't recognize joystick/gamepad

### ISSUE THREAD

**Issue by jackunity69** [igo95862/bubblejail#108]
### Description
I already check the joysticks and gamepads after configuring the instance in the GUI, steam does recognize my gamepad but the game is not. Do you know how to solve this? Thank you.

```
[common]
executable_name = "/usr/bin/steam"
share_local_time = false
filter_disk_sync = false
dbus_name = ""

[x11]

[wayland]

[network]

[pulse_audio]

[home_share]
home_paths = [
    "/home/user/.local/share/Steam/steamapps",
]

[direct_rendering]
enable_aco = false

[systray]

[joystick]

[root_share]
paths = []
read_only_paths = [
    "/sys/devices/system/cpu",
]

[pipewire]

```

**Comment by igo95862** [comment]
Hello @jackunity69 

I would recommend spawning a terminal inside the sandbox. For example using alacritty: `bubblejail run steam alacritty`

Using terminal you can use the `jstest` to see if any controllers are detected. (or any other equivalent programm that tests controllers)

The current controller algorithm is very error prone and should be reworked using udev. See #16

**Comment by igo95862** [comment]
> Maybe I'll try not to share home directory and download the game inside the sandbox.

I doubt that it will work.

The biggest issue will be if the game uses udev to enumerate controllers. In that case only passing entire udev socket would work.

**Comment by igo95862** [comment]
By the way does the Steam itself recognize the gamepad? I think if you go to options there should be gamepad options.

Also try disabling the Steam Input.

**Comment by jackunity69** [comment]
Yeah I've done that, still no works with the game. Steam does recognize the gamepad.

**Comment by odomingao** [comment]
I have this issue as well. Works fine with sandboxed native emulators, however no WINE game will recognize the controller.

**Comment by igo95862** [comment]
Probably something changed about how wine detects controllers. This needs to be investigated.

Hmm... Do an old Proton versions detect controllers? They definitely worked for me back when I started bubblejail in 2019.

**Comment by odomingao** [comment]
I couldn't find any game that would run with older Proton (6 and below) - if anyone knows of a game that is known to work, I'll test and report back.

Steam itself now also doesn't detect my controller anymore, though it used to.

**Comment by odomingao** [comment]
> > Maybe I'll try not to share home directory and download the game inside the sandbox.
> 
> I doubt that it will change anything.
> 
> The biggest issue will be if the game uses udev to enumerate controllers. In that case only passing entire udev socket would work.

How can I do this?

**Comment by igo95862** [comment]
It looks like if `/.flatpak-info` file is presented it will disable udev: https://github.com/ValveSoftware/wine/blob/f80ddae9e217f2bad68455891371fa0f524451e7/dlls/winebus.sys/bus_udev.c#L2103

**Comment by odomingao** [comment]
> It looks like if `/.flatpak-info` file is presented it will disable udev: https://github.com/ValveSoftware/wine/blob/f80ddae9e217f2bad68455891371fa0f524451e7/dlls/winebus.sys/bus_udev.c#L2103

hmm, I tried to create the file (`/.flatpak-info`) and pass it to the sandbox. Didn't seem to work, as I still get the same warning in the wine shell

**Comment by igo95862** [comment]
So creating the empty `/.flatpak-info` will actually cause games to fail to launch:

```
pressure-vessel-wrap[2648]: W: Running under Flatpak, unknown version
pressure-vessel-wrap[2648]: E: pressure-vessel (SteamLinuxRuntime) cannot be run in Flatpak 1.10.x or older. For Proton 5.13+, unofficial community builds that do not use pressure-vessel are available.
```

I tested several games both native and in proton experimental and the controller worked in all of them.

Not sure if it my controller or my selection of games but I didn't encounter any issues.

**Comment by igo95862** [comment]
Octopath Traveler did not want to use controller at first but as soon as I disabled the Steam Input it worked.

**Comment by odomingao** [comment]
Yep, it seems to work now on Steam (tested with DSR, Proton Experimental). They must have fixed it.

**Comment by odomingao** [comment]
@jackunity69 make sure to test with Proton Experimental. 9 wasn't working for me, but experimental did;

**Comment by jackunity69** [comment]
@odomingao yeah already did that but still though, I just run the game on vm now. though I still using bubblejail on kb/mouse game. I'll try other game in future to test out if it's the game problem or wayland/hyprland issue. Thank you guys.
